inputs.ping: Old/default "exec" method segfaults

I just upgraded to Telegraf 1.17.1 and it segfaults for me now due to a bug in inputs.ping. It looks like a new "native" method of pinging was introduced in this release, but for compatibility the default is to use the old "exec" method. However, the old exec method seems to segfault. I tested the new native method and that works. Would be great if the old method could be fixed for compatibility. Thank you!

### Relevant telegraf.conf:
<!-- Place config in the toml code section. -->
```toml
[[inputs.ping]]
   urls = ["www.google.com", "www.apple.com", "www.redacted.com", "ams-ix.net"]

```

### System info:

Telegraf 1.17.1 on Ubuntu 18.04 and 16.04

### Docker

No Docker.

### Steps to reproduce:

1. Use the config above
2. Run telegraf. It will segfault when it runs the ping check. See log below.

### Expected behavior:
 
It used to work and ping the hosts successfully.

### Actual behavior:

It segfaults!

### Additional info:

Log:
```
Jan 28 05:34:10 host telegraf[23893]: panic: runtime error: invalid memory address or nil pointer dereference
Jan 28 05:34:10 host telegraf[23893]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b2542a]
Jan 28 05:34:10 host telegraf[23893]: goroutine 80 [running]:
Jan 28 05:34:10 host telegraf[23893]: github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).pingToURL(0xc000170380, 0xc000222a21, 0xa, 0x34ae180, 0xc000202ec0)
Jan 28 05:34:10 host telegraf[23893]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping_notwindows.go:22 +0x18a
Jan 28 05:34:10 host telegraf[23893]: github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather.func1(0xc000170380, 0x34ae180, 0xc000202ec0, 0xc000222a21, 0xa)
Jan 28 05:34:10 host  telegraf[23893]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:138 +0x9b
Jan 28 05:34:10 host  telegraf[23893]: created by github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather
Jan 28 05:34:10 host telegraf[23893]:         /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:131 +0xd2
```

All of my site monitoring is down, since telegraf stopped working on all instances due to this ping bug. It crashes and after a few restarts it stops crashing but does nothing.

My config is using the default "exec" mode, and I a duplicating the parameters in the command line because after introducing '-4' to ping only ipv4 addresses it ignores them.
```
[[inputs.ping]]
  urls = [.....]
  count = 5
  ping_interval = 0.5
  timeout = 1.0
  deadline = 8
  arguments = ["-4","-c","5","-i","0.5","-W","1","-w","8"]
```
The problem affects all architectures running latest debian stable.
intel and arm examples of the crash:
```
2021-01-28T09:36:30Z I! Tags enabled: host=pi-base
2021-01-28T09:36:30Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"pi-base", Flush Interval:10s
2021-01-28T09:36:30Z I! [inputs.socket_listener] Listening on udp://127.0.0.1:4444
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x130c860]
goroutine 100 [running]:
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).pingToURL(0x51f3950,0x5523281, 0x7, 0x27a4828, 0x500a5a0)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping_notwindows.go:22 +0x11c0, 0x27a4828, 0x500a5a0, 0x5523281, 0x7)
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather.func1(0x51f3950, 0x27a4828, 0x500a5a0, 0x5523281, 0x7)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:138 +0xc0
created by github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:131 +0x9
```
```
2021-01-28T09:33:50Z I! Starting Telegraf 1.17.1
2021-01-28T09:33:50Z I! Loaded inputs: cpu dns_query docker exec (2x) file influxdb ntpq ping socket_listener
2021-01-28T09:33:50Z I! Loaded aggregators:
2021-01-28T09:33:50Z I! Loaded processors:
2021-01-28T09:33:50Z I! Loaded outputs: influxdb
2021-01-28T09:33:50Z I! Tags enabled: host=nuc
2021-01-28T09:33:50Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"nuc", Flush Interval:10s
2021-01-28T09:33:50Z I! [inputs.socket_listener] Listening on udp://127.0.0.1:4444
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b2542a]
goroutine 128 [running]:
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).pingToURL(0xc0001f8380, 0xc000211001, 0x7, 0x34ae180, 0xc000206760)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping_notwindows.go:22 +0x18a
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather.func1(0xc0001f8380, 0x34ae180, 0xc000206760, 0xc000211001, 0x7)
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:138 +0x9b
created by github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather
        /go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:131 +0xd2
```

same issue here. Just updated and now it's broken. My server metric monitoring still works but Ping and SNMP broke.

```2021-01-28T14:25:27Z I! Starting Telegraf 1.17.1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b2542a]

goroutine 16 [running]:
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).pingToURL(0xc00018c380, 0xc00031fc41, 0xb, 0x34ae180, 0xc00053a000)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping_notwindows.go:22 +0x18a
github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather.func1(0xc00018c380, 0x34ae180, 0xc00053a000, 0xc00031fc41, 0xb)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:138 +0x9b
created by github.com/influxdata/telegraf/plugins/inputs/ping.(*Ping).Gather
	/go/src/github.com/influxdata/telegraf/plugins/inputs/ping/ping.go:131 +0xd2```

--- a/plugins/inputs/ping/ping.go
+++ b/plugins/inputs/ping/ping.go
@@ -4,6 +4,7 @@ import (
 	"errors"
 	"fmt"
 	"math"
+	"os/exec"
 	"runtime"
 	"strings"
 	"sync"
@@ -11,6 +12,7 @@ import (
 
 	"github.com/go-ping/ping"
 	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/inputs"
 )
 
@@ -311,9 +313,21 @@ func (p *Ping) Init() error {
 	return nil
 }
 
+func hostPinger(binary string, timeout float64, args ...string) (string, error) {
+	bin, err := exec.LookPath(binary)
+	if err != nil {
+		return "", err
+	}
+	c := exec.Command(bin, args...)
+	out, err := internal.CombinedOutputTimeout(c,
+		time.Second*time.Duration(timeout+5))
+	return string(out), err
+}
+
 func init() {
 	inputs.Add("ping", func() telegraf.Input {
 		p := &Ping{
+			pingHost:     hostPinger,
 			PingInterval: 1.0,
 			Count:        1,
 			Timeout:      1.0,

--- a/plugins/inputs/ping/ping_test.go
+++ b/plugins/inputs/ping/ping_test.go
@@ -13,6 +13,7 @@ import (
 	"time"
 
 	"github.com/go-ping/ping"
+	"github.com/influxdata/telegraf/plugins/inputs"
 	"github.com/influxdata/telegraf/testutil"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
@@ -259,6 +260,21 @@ func TestPingGather(t *testing.T) {
 
 	tags = map[string]string{"url": "influxdata.com"}
 	acc.AssertContainsTaggedFields(t, "ping", fields, tags)
+}
+
+func TestPingGatherIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode, retrieves systems ping utility")
+	}
+
+	var acc testutil.Accumulator
+	p, ok := inputs.Inputs["ping"]().(*Ping)
+	require.True(t, ok)
+	p.Urls = []string{"localhost", "influxdata.com"}
+	err := acc.GatherError(p.Gather)
+	require.NoError(t, err)
+	require.Equal(t, 0, acc.Metrics[0].Fields["result_code"])
+	require.Equal(t, 0, acc.Metrics[1].Fields["result_code"])
 }
 
 var lossyPingOutput = `

